auslib/web/admin/swagger/api.yaml

```diff
@@ -0,0 +1,441 @@
+      consumes:
+        - application/json
+      produces:
+        - text/html
+      deprecated: false
+      operationId: auslib.web.admin.views.mapper.rules_post
+      parameters:
+        - name: rule
+          in: body
+          description: New Rule object data
+          required: true
+          schema:
+            allOf:
+              - $ref: '#/definitions/CSRFModel'
+              - $ref: '#/definitions/RulesBase'
+
+            # will fail testMissingField test in test_rules.py if test is not modified.
+            # Because only outputs first missing field : backgroundRate
+            # ignores further check i.e. check for priority and returns error.
+            required:
+              - backgroundRate
+              - priority
+
+      responses:
+        '200':
+          description: Created a new rule
+          schema:
+            type: string
+            example: "514"
+        '400':
+          description: Requires remote username and password in basic authentication
+          schema:
+            type: string
+            example: "{'error': 'invalid form values'}"
+          examples:
+            application/json: "{'error': 'Invalid Form Values'}"
+        default:
+          description: Requires remote username and password in basic authentication
+          examples:
+            text/html: "Error: REMOTE_USERNAME required"
+
+
+definitions:
+  CSRFModel:
+    title: CSRF Token
+    description: CSRF Token in POST
+    type: object
+
+    #Existing Test cases fail when we don't provide this value. Hence keep it out of required section.
+    #Make it optional.
+    #required:
+    #- csrf_token
+    properties:
+      csrf_token:
+        type: string
+        example: "1491342563##c4e6fef0b978e6c89af9ff1015e67b9ca7c45d14"
+
+  RulesBaseGET:
+    title: GET Operations on Rules Base
+    description: "Sample description for /rules GET"
+    allOf:
+      - $ref: "#/definitions/RulesBase"
+    required:
+      - alias
+      - backgroundRate
+      - buildID
+      - buildTarget
+      - channel
+      - comment
+      - data_version
+      - distVersion
+      - distribution
+      - fallbackMapping
+      - headerArchitecture
+      - locale
+      - mapping
+      - osVersion
+      - priority
+      - product
+      - rule_id
+      - systemCapabilities
+      - update_type
+      - version
+      #- whitelist
+
+  RulesBase:
+    title: Rules object definition
+    description: Balrog's Rule Attributes
+    type: object
+
+    required:
+      - update_type # Required during both POSTing and GET(fetch) Operations
+
+    #I was initially hopeful that x-nullable would serve the purpose at class level to define
+    # fields that can expect null values, but my research so
+    # far seems to show that currently x-nullable works at the parameter level but not
+    # within an object/model. Instead declaring 'type' as array like [string, null] or [integer, null]
```

Comment:
I was still trying to still experiment with x-nullable at object/model level. Doesn't work for current version of connexion (1.1.9) , so will remove that from all the fields. Also  declaring 'type' as array like [string, null] is in compliance with jsonschema. [Jsonschema : type](https://spacetelescope.github.io/understanding-json-schema/reference/type.html)